jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1687,6 +1687,224 @@ ecma_builtin_typedarray_prototype_find_index (ecma_value_t this_arg, /**< this a
+    ecma_number_t target_element = ecma_get_typedarray_element (typedarray_buffer_p + start,
+                                                                class_id);
+
+    ecma_set_typedarray_element (typedarray_buffer_p + target, target_element, class_id);
+
+    if (direction < 0)
+    {
+      start -= element_size;
+      target -= element_size;
+    }
+    else
+    {
+      start += element_size;
+      target += element_size;
+    }
+
+    count--;
+  }
+
+  return ecma_copy_value (this_arg);
+} /* ecma_builtin_typedarray_prototype_copy_within */
+
+/**
+ * The %TypedArray%.prototype object's 'slice' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 22.2.3.23
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_slice (ecma_value_t this_arg, /**< this argument */
+                                         const ecma_value_t args[], /**< arguments list */
+                                         ecma_length_t args_number) /**< number of arguments */
+{
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+
+  ecma_object_t *typedarray_p = ecma_get_object_from_value (this_arg);
+  uint32_t length = ecma_typedarray_get_length (typedarray_p);
+  lit_magic_string_id_t class_id = ecma_object_get_class_name (typedarray_p);
+  lit_utf8_byte_t *typedarray_buffer_p = ecma_typedarray_get_buffer (typedarray_p);
+  uint8_t shift = ecma_typedarray_get_element_size_shift (typedarray_p);
+  uint8_t element_size = (uint8_t) (1 << shift);
+  uint32_t start = 0;
+  uint32_t end = length;
+
+  if (args_number > 0)
+  {
+    ecma_value_t start_value = ecma_builtin_get_uint32_value_from_argument (args[0], length, &start, false);
+
+    if (ECMA_IS_VALUE_ERROR (start_value))
+    {
+      return start_value;
+    }
+
+    if (args_number > 1)
+    {
+      ecma_value_t end_value = ecma_builtin_get_uint32_value_from_argument (args[1], length, &end, true);
+
+      if (ECMA_IS_VALUE_ERROR (end_value))
+      {
+        return end_value;
+      }
+    }
+  }
+
+  int32_t distance = (int32_t) (end - start);
+  uint32_t count = distance > 0 ? (uint32_t) distance : 0;
+
+  ecma_value_t new_typedarray = ecma_op_create_typedarray_with_type_and_length (typedarray_p, count);
+
+  if (ECMA_IS_VALUE_ERROR (new_typedarray))
+  {
+    return new_typedarray;
+  }
+
+  ecma_object_t *new_typedarray_p = ecma_get_object_from_value (new_typedarray);
+  lit_magic_string_id_t new_class_id = ecma_object_get_class_name (new_typedarray_p);
+  lit_utf8_byte_t *new_typedarray_buffer_p = ecma_typedarray_get_buffer (new_typedarray_p);
+
+  if (count > 0)
+  {
+    ecma_length_t src_byte_offset = ecma_typedarray_get_offset (typedarray_p);
+    uint32_t target_byte_index = 0;
+    uint32_t src_byte_index = (start * element_size) + src_byte_offset;
+
+    while (target_byte_index < count * element_size)
```

Comment:
The source and target buffers will always have the same type, since the target is cloned from the source. We could do a single memcpy instead of the while loop.